ens/utils.py

```diff
@@ -102,11 +113,41 @@ def normalize_name(name: str) -> str:
         name = name.decode('utf-8')
 
     try:
-        return idna.uts46_remap(name, std3_rules=True)
+        return idna.uts46_remap(name, std3_rules=True, transitional=False)
     except idna.IDNAError as exc:
         raise InvalidName(f"{name} is an invalid name, because {exc}") from exc
 
 
+def ens_encode_name(name: str) -> bytes:
+    """
+    Encode a name according to DNS standards specified in section 3.1 of RFC1035 with the following
+    exceptions:
+
+        - There is no limit on the total length of the encoded name while still keeping the length
+        limit on labels as the DNS-standard of 63.
+
+        - Return a single 0-octet, b'\x00', if empty name.
+    """
+    if is_empty_name(name):
+        return b'\x00'
+
+    normalized_name = normalize_name(name)
+
+    labels = normalized_name.split('.')
+    labels_as_bytes = [to_bytes(text=label) for label in labels]
+
+    # raises if len(label) > 63:
+    for index, label in enumerate(labels):
+        if len(label) > 63:
```

Comment:
DNS has a limit of 63 octets per label, but the limit in ENS is 255.

Reply:
From what I saw in ENSIP-10, the only deviations from the DNS encoding is that the zero octet is returned for the empty string and that "there is no limit on the total length of the encoded name". Does the 63 size limit not matter here then? Should it be 255 for labels and no limit on total length?

I just checked out the [ethers.js](https://github.com/ethers-io/ethers.js/blob/33a029e457320a226c68a01f4cfa2d110125a8b8/packages/hash/src.ts/namehash.ts#L49-L57) implementation and it looks like they may be using a limit of 63 as well.

Reply:
DNS encoding also specifies the zero octet for the empty string. The 63 byte limit is a limitation of the DNS protocol, not the encoding mechanism, and we don't enforce the same limit. 255 should be the limit for labels, with no limit on total length.

Reply:
Yep, makes sense. I'll push an update. Thanks again 👌